lib/mox.ex

```diff
@@ -524,7 +524,9 @@ defmodule Mox do
 
     for behaviour <- behaviours,
         {fun, arity} <- behaviour.behaviour_info(:callbacks) do
-      stub(mock, fun, :erlang.make_fun(module, fun, arity))
+      if function_exported?(mock, fun, arity) do
```

Comment:
Something along these lines:

    for behaviour <- behaviours,
        optional = behaviour.behaviour_info(:optional_callbacks),
        {fun, arity} <- behaviour.behaviour_info(:callbacks) do
      if function_exported?(mock, fun, arity) or {fun, arity} not in optional do

We should add a test that `stub_with` fails when some non-optional callback is missing.

Reply:
I agree that you should always implement the full behaviour in the Stub module, but enforcing this would possibly break a lot of current usage.  Currently this test succeeds:

```
    defmodule PartialCalcImplementation do
      @behaviour Calculator
      def add(x, y), do: x + y
    end

    test "partial" do
      in_all_modes(fn ->
        stub_with(CalcMock, PartialCalcImplementation)
        assert CalcMock.add(1, 2) == 3

        assert_raise UndefinedFunctionError, fn ->
          assert CalcMock.mult(3, 4)
        end
      end)
    end
```

Reply:
I'd be happy to change that you can only stub full behaviour implementations of course.